src/commands/deploy/deploy.ts

```diff
@@ -155,6 +156,7 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
     // tslint:disable-next-line:no-floating-promises
     node.promptToSaveDeployDefaults(deployContext, deployContext.workspace.uri.fsPath, deployContext.deployFsPath);
 
+    await setPreDeployTaskForDotnet(deployContext);
```

Comment:
I feel like I would expect this prompt _before_ the "Are you sure you want to deploy?" prompt. Not 100% sure on this though

Reply:
Yeah, I agree with that now that you bring it up.